programs/stake/src/stake_instruction.rs

```diff
@@ -3250,6 +3354,16 @@ mod tests {
 
         // should pass, withdrawing initialized account down to minimum balance
         process_instruction(
+            &feature_set,
+            &serialize(&StakeInstruction::Withdraw(stake_lamports)).unwrap(),
+            transaction_accounts.clone(),
+            instruction_accounts.clone(),
+            Ok(()),
+        );
+
+        // should fail, withdrawing account down to only rent-exempt reserve
+        process_instruction(
+            &feature_set,
             &serialize(&StakeInstruction::Withdraw(stake_lamports)).unwrap(),
```

Comment:
Hmmm, looks like maybe this specific Withdraw was a mishandled rebase conflict. See #24670, which changed this part.

Also, Withdrawing 1 lamport from an Initialized stake account is OK, yes? (At least that's what the current behavior allows.)

Reply:
I thought the processed instruction had a side effect so I thought that a subsequent withdrawal would push it below the remaining balance. Maybe I misinterpreted that. Feel free to patch how you see fit.

Reply:
Yes, that was a change when process_instruction moved from KeyedAccounts to BorrowedAccounts. Now `process_instruction()` returns a vector of accounts, post processing. Those need to be used for subsequent instructions, because the original accounts are unmodified.